madlib.js

```diff
@@ -0,0 +1,154 @@
+	}
+	else{
+		var title = "Hey player, choose a story first";
+		var story = "-_-";
+	}
+
+	//Output elements
+	var h2 = document.createElement("h2");
+	h2.innerHTML = title;
+	h2.className = "title";
+    var out = document.getElementById("out");
+    out.className = "out ostyle";
+    out.innerHTML = story;
+    out.insertBefore( h2, out.firstChild);
+
+}
+
+function getInput(wordType,wordsNeeded){
+
+	inputs = document.getElementById(wordType).value;
+
+	//if no user inputs, make inputArray blank
+	if(inputs === ""){
+		var inputArray = [];
+	}else{
+		var inputArray = inputs.split(', ');
+	}
+
+	console.log("Before inputArray", inputArray);
+	var nounRandArray = [];
+	var nounRand;
+	var verbRandArray = [];
+	var verbRand;
+	var adjectiveRandArray = [];
+	var adjectiveRand;
+	var adverbRandArray = [];
+	var adverbRand;
+	var verbingRandArray = [];
+	var verbingRand;
+
+	//if default values are needed, load the defaults
+	while(inputArray.length < wordsNeeded){
+
+		if(wordType === "noun"){
+			//make sure there are no duplicates
+			do{
+				nounRand = Math.floor(Math.random()*(noun.length));
+			}while(nounRandArray.indexOf(nounRand) !== -1)
+			nounRandArray.push(nounRand);
+			console.log("nounRandArray",nounRandArray);
+			inputArray.push(noun[nounRand]);
+
+		}else if(wordType === "verb"){
+			//make sure there are no duplicates
+			do{
+				verbRand = Math.floor(Math.random()*(verb.length));
+			}while(verbRandArray.indexOf(verbRand) !== -1)
+			verbRandArray.push(verbRand);
+			console.log("verbRandArray",verbRandArray);
+			inputArray.push(verb[verbRand]);
+
+		}else if(wordType === "adjective"){
+			//make sure there are no duplicates
+			do{
+				adjectiveRand = Math.floor(Math.random()*(adjective.length));
+			}while(adjectiveRandArray.indexOf(adjectiveRand) !== -1)
+			adjectiveRandArray.push(adjectiveRand);
+			console.log("adjectiveRandArray",adjectiveRandArray);
+			inputArray.push(adjective[adjectiveRand]);
+
+		}else if(wordType === "adverb"){
+			//make sure there are no duplicates
+			do{
+				adverbRand = Math.floor(Math.random()*(adverb.length));
+			}while(adverbRandArray.indexOf(adverbRand) !== -1)
+			adverbRandArray.push(adverbRand);
+			console.log("adverbRandArray",adverbRandArray);
+			inputArray.push(adverb[adverbRand]);
+
+		}else if(wordType === "verbing"){
+			//make sure there are no duplicates
+			do{
+				verbingRand = Math.floor(Math.random()*(verbing.length));
+			}while(verbingRandArray.indexOf(verbingRand) !== -1)
+			verbingRandArray.push(verbingRand);
+			console.log("verbingRandArray",verbingRandArray);
+			inputArray.push(verbing[verbingRand]);
+
+		}
+	}
+
+	var finalArray = [];
+	console.log("After inputArray", inputArray);
+	var rando;
+	var randoArray = [];
+
+	for(i = 0; i < wordsNeeded; i++) {
+
+		//make sure there are no duplicates in final array of random values
+		do{
+			rando = Math.floor(Math.random()*(wordsNeeded));
+		}while(randoArray.indexOf(rando) !== -1)
+		randoArray.push(rando);
```

Comment:
Good job at randomizing the input twice. You are the only one who actually did it twice!